src/main/java/com/currencycloud/client/CurrencyCloudClient.java

```diff
@@ -99,24 +97,9 @@ String getAuthToken() {
      * @throws IllegalStateException    if onBehalfOf is already set (nested call to this method)
      * @throws IllegalArgumentException if onBehalfOf is in illegal format
      */
-    public void onBehalfOfDo(String contactId, Runnable work)
+    public void onBehalfOfDo(String contactId, OnBehalfRunnable work)
```

Comment:
New setup does not seem to be backward compatible. There is probably no way around this because the old setup was not thread safe. However, we will need to send out a communication to at least the people we know use the Java SDK and possible everyone else too.

Reply:
Given it is not backward compatible is there a more elegant solution than overlaoding every method? Might just be a Java limitation.